jax/interpreters/pxla.py

```diff
@@ -612,23 +612,19 @@ def parallel_callable(fun, backend, axis_name, axis_size, global_axis_size,
   if devices is not None and len(devices) == 0:
     raise ValueError("'devices' argument to pmap must be non-empty, or None.")
 
+  inner_pmap = len(_thread_local_state.dynamic_axis_env) == 0
```

Comment:
I'm not too familiar with what's going on in DynamicAxisEnv so please override me, but isn't this reversed?
We're in an inner pmap if there are already items in `dynamic_axis_env`; i.e. len > 0?

Reply:
Gah you're right! I guess it still worked because the test script has `global_axis_size` set for both pmaps, and I'm guessing the inner pmap never makes it to the `num_local_shards` check because partial eval probably yields a trivial computation that isn't actually compiled. I fixed it, and FWIW the test script still passes...